src/hiccup/compiler.clj

```diff
@@ -13,17 +14,33 @@
 (defn- end-tag []
   (if (xml-mode?) " />" ">"))
 
+(defn- render-attr-value [value]
+  (cond
+    (string? value)
+    (util/escape-html (util/as-str value))
+    (map? value)
+    (str
+      (->> value
+           (mapv (fn [[k v]] (str (util/as-str k) ":" (util/as-str v))))
+           sort
+           (string/join ";"))
+      ";")
+    (sequential? value)
+    (string/join " " value)
```

Comment:
Yes, though you could just factor the `escape-html` out entirely, since every condition will require it.